modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -332,29 +296,71 @@ ThreadPool getThreadPool() {
         return threadPool;
     }
 
-    static class Stats {
+    enum Operation {
+        GET_OBJECT("GetObject"),
+        LIST_OBJECTS("ListObjects"),
+        PUT_OBJECT("PutObject"),
+        PUT_MULTIPART_OBJECT("PutMultipartObject"),
+        DELETE_OBJECTS("DeleteObjects"),
+        ABORT_MULTIPART_OBJECT("AbortMultipartObject");
+
+        private final String key;
+
+        String getKey() {
+            return key;
+        }
 
-        final AtomicLong listCount = new AtomicLong();
+        Operation(String key) {
+            this.key = key;
+        }
+    }
 
-        final AtomicLong getCount = new AtomicLong();
+    record StatsKey(Operation operation, OperationPurpose purpose) {}
 
-        final AtomicLong putCount = new AtomicLong();
+    class StatsCollectors {
+        final Map<StatsKey, IgnoreNoResponseMetricsCollector> collectors = new ConcurrentHashMap<>();
 
-        final AtomicLong postCount = new AtomicLong();
+        StatsCollectors() {}
 
-        final AtomicLong deleteCount = new AtomicLong();
+        RequestMetricCollector getMetricCollector(Operation operation, OperationPurpose purpose) {
+            return collectors.computeIfAbsent(new StatsKey(operation, purpose), k -> buildMetricCollector(k.operation()));
+        }
 
-        final AtomicLong abortCount = new AtomicLong();
+        Map<String, Long> statsMap() {
+            final Map<String, Long> stats = Arrays.stream(Operation.values()).collect(Collectors.toMap(e -> e.key, e -> 0L));
+            collectors.forEach((k0, v) -> stats.compute(k0.operation().getKey(), (k1, c) -> Objects.requireNonNull(c) + v.counter.get()));
+            return Map.copyOf(stats);
```

Comment:
I think it is easier to do it declarative style:
```suggestion
            return collectors.entrySet().stream().collect(groupingBy(
                entry -> entry.getKey().operation().getKey(),
                summingLong(entry -> entry.getValue().counter.get())
            ));
```
Please note, this will skip operations that are not in `collectors` and this produce mutable map. I think there is an overloaded version available to collect to immutable map.

Reply:
> this will skip operations that are not in collectors 

Unfortunately this is the reason why the aggregation is a bit more involved because we need operations that are _not_ in the collectors for BWC.

Reply:
Oh, I see. This could be solved by:
```
return Stream.concat(
    Stream.of(Operation.values()).map(value -> Map.entry(value.getKey(), 0L)),
    collectors.entrySet().stream().map(entry -> Map.entry(entry.getKey().operation.getKey(), entry.getValue().counter.get()))
).collect(groupingBy(
    Map.Entry::getKey,
    summingLong(Map.Entry::getValue)
));
```
but at this point I do not think if this reads better nor cheaper to execute

Reply:
Yeah this is the version I started with locally more or less. I didn't use it since I thought the nested stream of entrySet maybe frowned up. I think it could be an interesting discussion separately to sorta decide what kinda stream operations are undesirable. For this particular use case, I'll leave it as is if you don't mind.